src/overseer/core.clj

```diff
@@ -85,6 +85,27 @@
   []
   (quot (System/currentTimeMillis) 1000))
 
+; Core functions encompassing the system's interaction with storage backends
+; These functions effectively represent the finite state machine transitions
+; that a job goes through:
+;
+;      unstarted ---------+
+;       |  ^              |
+;       |  |              |
+;       v  |              |
+;      started -----------+
+;      /     \            |
+;     /        \          |
+; finished   failed   aborted
+;
+; The system internally handles the transitions:
+;   (unstarted -> started)
+;   (started -> finished)
+;   (started -> failed)
+;
+; Callers can force transitions using the functions in `overseer.api`:
+;   (started -> unstarted)
+;   (* -> aborted)
```

Comment:
Heh. Yes, this needs to be scoped down. The code doesn't (currently) examine the current status before the transition though.